lib/Backend/IRBuilder.cpp

```diff
@@ -118,7 +118,7 @@ IRBuilder::DoBailOnNoProfile()
         return false;
     }
 
-    if (m_func->GetTopFunc()->GetJITFunctionBody()->IsCoroutine())
+    if (m_func->GetTopFunc()->GetJITFunctionBody()->IsCoroutine() && !m_func->IsLoopBody())
```

Comment:
BailOnNoProfile caused problems when put in a loop containing a Yield; not a problem when jitting loop bodies that cannot contain yield.